### DISCUSSION TITLE
Ability to embed Swagger/OpenAPI spec

### DISCUSSION THREAD

**Discussion by kamal0808** [outline/outline#2521] [Ideas]
**First**, really awesome project. Love it! Thanks for making this **open-source**.

**Second,** OpenAPI is being used as one of the most convenient ways to share API docs and sandbox environments. We at tech companies share a lot of documents that look like below:
![D76DA613-2C1C-44D5-9697-37AD5071B58E](https://user-images.githubusercontent.com/15616412/131656157-e1127c52-66f1-4ca1-b8f4-405455102980.jpeg)

Only if they converted themselves to a UI that the ops team and other non-tech people can get their hands on, it would be amazing.

_I’m okay contributing to this feature, or if someone else works on it. Do create a discussion below if whoever is reading this has some ideas around how/whether to have this feature._

Here’s how Swagger’s own editor-to-UI looks like:

![F18CDCFF-CE5F-4062-B79C-985C5F5D51D9](https://user-images.githubusercontent.com/15616412/131656580-23fd7f88-0a80-4995-8406-c5e0123ae7f9.jpeg)

**Comment by JrmyDev** [comment]
Would be so great to have this feature !!

**Comment by carun** [comment]
Can we have this please? It's been 3 years since this request was raised.

**Comment by almereyda** [comment]
The API documentation including OpenAPI schema download is available from https://www.getoutline.com/developers.

There is also a freely licensed repository at [outline/openapi](https://github.com/outline/openapi) with resources and tooling.

**Reply by LEI** [reply]
Unless I am mistaken this is about embeding OpenAPI schemas to replace an external Swagger UI https://petstore.swagger.io/.

The Outline API documentation seems to use https://scalar.com/ for the "try it out" feature.

I am following this discussion in case a specific integration is recommended, even if it is a [simple HTML outline](https://gohugo.io/functions/openapi3/unmarshal/).

**Reply by almereyda** [reply]
I see, I was mistaken.
